### PR TITLE
Added filter to MCQs in create/edit drive form

### PR DESCRIPTION
**Pull request by somnath77**
https://trello.com/c/1uLemALq

### CODE REVIEW

File: src/modules/admin/createDrive/CreateDriveComponent/addQuestionCmponent.jsx [hunk 1]
```diff
@@ -25,15 +41,9 @@ const AddQuestionComponent = (props) => {
...
-  const selected = [];
-  selectedQuestions.map((e) => {
-    const question = questions.find((x) => x.id === e);
-    return (question && selected.push(question));
-  });
-
   const questionOptions = [];
   questions.map((e) => {
-    const flg = (selected.find((x) => x.id === e.id));
+    const flg = (selectedQuestionsDetails.find((x) => x.id === e.id));
```
Comment (Durgesh00) [inline src/modules/admin/createDrive/CreateDriveComponent/addQuestionCmponent.jsx:46]: Can have a more specific name.

File: src/modules/admin/createDrive/CreateDriveComponent/addQuestionCmponent.jsx [hunk 2]
```diff
@@ -42,17 +52,66 @@ const AddQuestionComponent = (props) => {
       <div className='new-drive-content-head d-flex px-3 align-items-center'>
         <label className='mb-0'>Add MCQs to Drive</label>
         <div className='d-flex ml-auto align-items-center'>
-          <Icon name='search' className='font-weight-bold' />
+          <Icon name='search' onClick={handleFilterQuestionBox} className='font-weight-bold search-icon' />
           <span className='count-box px-3 py-2 ml-3'>
-            {selectedQuestions.length} selected
+            {selectedQuestionsDetails.length} selected
```
Comment (Durgesh00) [inline src/modules/admin/createDrive/CreateDriveComponent/addQuestionCmponent.jsx:57]: good to calculate above the return statement.

File: src/redux/admin/questionsList/reducer.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  isQuestionsError: false,
+};
+
+const questionsReducer = produce((state = initialState, action = {}) => {
+  const { type, payload } = action;
+  switch (type) {
+    case QUESTIONS_LIST.QUESTIONS_LIST_REQUEST_ACTION:
+      state.isQuestionsLoading = true;
+      break;
+    case QUESTIONS_LIST.QUESTIONS_LIST_SUCCESS_ACTION:
```
Comment (Durgesh00) [inline src/redux/admin/questionsList/reducer.js:18]: good to get the keys at the top like:- const {QUESTIONS_LIST_REQUEST_ACTION,.QUESTIONS_LIST_SUCCESS_ACTION,.QUESTIONS_LIST_FAILURE_ACTION  } = QUESTIONS_LIST;
and use it afterwards in reducer and avoid the . operator

File: src/modules/admin/createDrive/CreateDriveComponent/addQuestionCmponent.jsx (hunk 1 above)
Comment (somnath77) [reply src/modules/admin/createDrive/CreateDriveComponent/addQuestionCmponent.jsx:46]: resolved

File: src/modules/admin/createDrive/CreateDriveComponent/addQuestionCmponent.jsx (hunk 2 above)
Comment (somnath77) [reply src/modules/admin/createDrive/CreateDriveComponent/addQuestionCmponent.jsx:57]: resolved

File: src/redux/admin/questionsList/reducer.js (hunk 3 above)
Comment (somnath77) [reply src/redux/admin/questionsList/reducer.js:18]: YES, made changes

**Review by Durgesh00 — APPROVED** [review]

**Pull request merged**